pkg/api/node/formatter.go

```diff
@@ -188,12 +188,25 @@ func (h ActionHandler) retryMaintenanceModeUpdate(nodeName string, updateFunc ma
 
 func (h ActionHandler) listUnhealthyVM(rw http.ResponseWriter, node *corev1.Node) error {
 	ndc := nodedrain.ActionHelper(h.nodeCache, h.virtualMachineInstanceCache, h.longhornVolumeCache, h.longhornReplicaCache)
-	vmList, err := ndc.FindAndListVM(node)
+
+	var respObj ListUnhealthyVM
+
+	nonMigrtableVMList, err := ndc.FindAndListNonMigratableVM(node)
 	if err != nil {
 		return err
 	}
 
-	var respObj ListUnhealthyVM
+	if len(nonMigrtableVMList) > 0 {
+		respObj.Message = "Please detach CDROM or ContainerDisk from following VMs:"
+		respObj.VMs = nonMigrtableVMList
+		rw.WriteHeader(http.StatusOK)
+		return json.NewEncoder(rw).Encode(&respObj)
```

Comment:
prefer to use `util.ResponseOKWithBody` instead.
https://github.com/ibrokethecloud/harvester/blob/06196d2d2ea33d036e740c025b3ec8d65fbec1ec/pkg/util/http.go#L19-L23

Reply:
`util.ResponseOKWithBody` absorbs the error during the json Marshalling or http response write. Do we still prefer to use this?

Reply:
I think exposure to response would be better, @guangbochen what do you think?

Reply:
I think we can keep the current path to expose bugs directly since it will be a composite call of this method on the UI, thanks.